src/http/client.rs

```diff
@@ -41,10 +42,12 @@ impl<'a> Client<'a> {
         // is to trap if we try and get the response more than once. The final
         // `?` is to raise the actual error if there is one.
         let res = res.get().unwrap().unwrap()?;
-        Ok(Response::try_from_incoming_response(
-            res,
-            self.reactor.clone(),
-        )?)
+        Response::try_from_incoming_response(res, self.reactor.clone())
+    }
+
+    /// HTTP GET request.
+    pub async fn get(&self, url: Url) -> Result<Response<IncomingBody>> {
+        self.send(Request::new(Method::Get, url)).await
     }
```

Comment:
Can we remove this? I understand this comes in handy when testing, but if we add this for the `Get` method we will want to add it for more methods too - and that ends up adding additional API surface.